packages/@react-spectrum/list/src/listview.css

```diff
@@ -23,60 +24,135 @@
   padding: 0;
   background-color: var(--spectrum-table-background-color, var(--spectrum-global-color-gray-50));
   outline: 0;
+  user-select: none;
 
-  [role="row"] {
+  .react-spectrum-ListView-row {
     outline: none;
   }
+
+  &.react-spectrum-ListView--emphasized {
+    .react-spectrum-ListViewItem {
+      /* common pseudoelement for box shadows */
+      &:after {
+        content: '';
+        display: block;
+        position: absolute;
+        inset-inline-start: 0;
+        inset-inline-end: 0;
+        inset-block-end: 0;
+        inset-block-start: 0;
+        z-index: 3;
+        pointer-events: none;
+      }
+
+      /* Box shadow for bottom border for non-selected rows that is immmediately above a selected row. */
+      &.is-next-selected {
+        &:after {
+          box-shadow: inset 0 -1px 0 0 var(--spectrum-global-color-blue-500);
```

Comment:
Reverted back to the 1px bottom box shadow border strategy (strategy used for the TableView) instead of the .5px top/bottom box shadows used previously due to https://github.com/adobe/react-spectrum/issues/3017. Hopefully this doesn't have the same zoom issues.

Unfortunately means that the keyboard focus indicator will look 1px off due to the top border being an inset box shadow of the row above:
current (1px inset bottom box shadow border strategy):
Note the blue focus indicator doesn't overlap the top border of row 2 
<img width="834" alt="image" src="https://user-images.githubusercontent.com/9661127/162544193-ed2ecff0-4ec5-4d3d-aa9a-12b08212f19f.png">

previous (.5 px top and bottom box shadow border stategy):
Note the blue focus indicator overlaps the top and bottom border of row 2, albeit with .5px extra grey border from the previous and preceeding rows box shadows
<img width="941" alt="image" src="https://user-images.githubusercontent.com/9661127/162544247-4be0df0f-7b73-47fa-a574-7703ddad3bf2.png">

Reply:
Perhaps to solve this ideally, the two rows should have a gap between them that is strictly reserved for the border? Or maybe the two rows should overlap and have the same inset border? Either way, I wanna punt this for a followup, don't wanna change too much for this PR